src/coreclr/nativeaot/Runtime/amd64/GC.asm

```diff
@@ -19,3 +19,20 @@ LEAF_ENTRY xmmYmmStateSupport, _TEXT
 LEAF_END xmmYmmStateSupport, _TEXT
 
         end
+
+;; extern "C" DWORD __stdcall zmmStateSupport();
+LEAF_ENTRY zmmStateSupport, _TEXT
+        mov     ecx, 0                  ; Specify xcr0
+        xgetbv                          ; result in EDX:EAX
+        and eax, 70H
+        cmp eax, 70H                    ; check OS has enabled ZMM state support
```

Comment:
I think this needs to be `E0H`. The bits we want to check are `XCR0[7:5]` where:
* XCR0[0] is associated with x87 state
* XCR0[1] is associated with SSE state
* XCR0[2] is associated with AVX state
* XCR0[4:3] are associated with MPX state
* XCR0[7:5] are associated with AVX-512 state
* XCR0[8] is reserved
* XCR0[9] is associated with PKRU state
* XCR0[63:10] isreserved.

Additionally, given that we technically need to check both `XCR0[7:5] == 111b` and `XCR0[2:1] == 11b`, it might be better to just check against `E6H` to ensure that any call to this checks the necessary bits and that we don't strictly require calling both `xmmYmmStateSupport` and `zmmStateSupport` independently.

Reply:
That is a bad mistake on my part. Thanks for catching it and pointing it out. I went with `0E6H` so what we check for all of `SSE AVX and AVX512` states. This reflected in the change in `EEJitManager::SetCpuInfo()`